pkg/search/proxy/framework/plugins/cluster/cluster.go

```diff
@@ -0,0 +1,126 @@
+package cluster
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	listcorev1 "k8s.io/client-go/listers/core/v1"
+
+	clusterlisters "github.com/karmada-io/karmada/pkg/generated/listers/cluster/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/search/proxy/framework"
+	"github.com/karmada-io/karmada/pkg/search/proxy/store"
+	"github.com/karmada-io/karmada/pkg/util/proxy"
+)
+
+const (
+	order = 2000
+)
+
+// Cluster proxies the remaining requests to member clusters:
+// - writing resources.
+// - or subresource requests, e.g. `pods/log`
+// We firstly find the resource from cache, and get the located cluster. Then redirect the request to the cluster.
+type Cluster struct {
+	Store         store.Store
+	ClusterLister clusterlisters.ClusterLister
+	SecretLister  listcorev1.SecretLister
+}
+
+var _ framework.Plugin = (*Cluster)(nil)
+
+// New creates an instance of Cluster
+func New(dep framework.PluginDependency) (framework.Plugin, error) {
+	secretLister := dep.KubeFactory.Core().V1().Secrets().Lister()
+	clusterLister := dep.KarmadaFactory.Cluster().V1alpha1().Clusters().Lister()
+
+	return &Cluster{
+		Store:         dep.Store,
+		ClusterLister: clusterLister,
+		SecretLister:  secretLister,
+	}, nil
+}
+
+// Order implements Plugin
+func (c *Cluster) Order() int {
+	return order
+}
+
+// SupportRequest implements Plugin
+func (c *Cluster) SupportRequest(request framework.ProxyRequest) bool {
+	return request.RequestInfo.IsResourceRequest && c.Store.HasResource(request.GroupVersionResource)
+}
+
+// Connect implements Plugin
+func (c *Cluster) Connect(ctx context.Context, request framework.ProxyRequest) (http.Handler, error) {
+	requestInfo := request.RequestInfo
+
+	if requestInfo.Verb == "create" {
```

Comment:
shoud disallow `update`?

Reply:
This is moved from original cluster.go. Changing logic is not the goal of this PR.

@ikaven1024

Reply:
> shoud disallow `update`?

`update` is supported. 
We don't support `create` because proxy doesn't know which cluster to create in.
https://github.com/karmada-io/website/blob/main/docs/userguide/globalview/proxy-global-resource.md#is-creating-supported